scylla-cdc/src/cdc_types.rs

```diff
@@ -29,15 +29,20 @@ impl SerializeCql for GenerationTimestamp {
     }
 }
 
-impl FromCqlVal<CqlValue> for GenerationTimestamp {
-    fn from_cql(cql_val: CqlValue) -> Result<Self, FromCqlValError> {
-        match cql_val {
-            CqlValue::Timestamp(val) => {
-                let timestamp = chrono::Duration::milliseconds(val.0);
-                Ok(GenerationTimestamp { timestamp })
-            }
-            _ => Err(FromCqlValError::BadCqlType),
-        }
+impl<'frame, 'metadata> DeserializeValue<'frame, 'metadata> for GenerationTimestamp {
+    fn type_check(typ: &ColumnType) -> Result<(), scylla::deserialize::TypeCheckError> {
+        <CqlTimestamp as DeserializeValue<'frame, 'metadata>>::type_check(typ)
+    }
+
+    fn deserialize(
+        typ: &'metadata ColumnType<'metadata>,
+        v: Option<scylla::deserialize::FrameSlice<'frame>>,
+    ) -> Result<Self, scylla::deserialize::DeserializationError> {
+        let timestamp = <CqlTimestamp as DeserializeValue<'frame, 'metadata>>::deserialize(typ, v)?;
+
+        Ok(GenerationTimestamp {
+            timestamp: chrono::Duration::milliseconds(timestamp.0),
+        })
```

Comment:
I'm curious, do you know if there is a reason GenerationTimestamp stores `timestamp` as `chrono::Duration` instead of `DateTime` / `NaiveDateTime`? `Duration` should be used to represent offset between two points in time, not a single point in time.

It would also simplify this a bit, because `DateTime` / `NaiveDateTime` are supported by Rust Driver.